File: Programming Basics with JS/T05 - For Loop/exercise/03. Histogram.js
function histogram(input) {
    let numbers = Number(input[0]);
    let count1 = 0;
    let count2 = 0;
    let count3 = 0;
    let count4 = 0;
    let count5 = 0;

    for (let i = 1; i <= numbers; i++) {
        if (Number(input[i]) < 200) {
            count1++;
        } else if (Number(input[i]) >= 200 && Number(input[i]) < 400) {
            count2++;
        } else if (Number(input[i]) >= 400 && Number(input[i]) < 600) {
            count3++;
        } else if (Number(input[i]) >= 600 && Number(input[i]) < 800) {
            count4++;
        } else if (Number(input[i]) >= 800) {
            count5++;
        }
    }

    let p1 = count1 / numbers * 100;
    let p2 = count2 / numbers * 100;
    let p3 = count3 / numbers * 100;
    let p4 = count4 / numbers * 100;
    let p5 = count5 / numbers * 100;

    console.log(`${p1.toFixed(2)}%`);
    console.log(`${p2.toFixed(2)}%`);
    console.log(`${p3.toFixed(2)}%`);
    console.log(`${p4.toFixed(2)}%`);
    console.log(`${p5.toFixed(2)}%`);
}
histogram(['3', '1', '2', '999']);
histogram(['7', '800', '801', '250', '199', '399', '599', '799']);
histogram(["9", "367", "99", "200", "799", "999", "333", "555", "111", "9"]);
histogram(["14", "53", "7", "56", "180", "450", "920", "12", "7", "150", "250", "680", "2", "600", "200"]);

File: Programming Basics with Go/T05 - For Loop/exercise/03. Histogram.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scan(&n)
	var count1, count2, count3, count4, count5 int = 0, 0, 0, 0, 0

	for number := 1; number <= n; number++ {
		var value float32
		fmt.Scan(&value)

		if value < 200 {
			count1++
		} else if value >= 200 && value <= 399 {
			count2++
		} else if value >= 400 && value <= 599 {
			count3++
		} else if value >= 600 && value <= 799 {
			count4++
		} else if value >= 800 {
			count5++
		}

	}

	fmt.Printf("%.2f%%\n", float32(count1) / float32(n) * 100)
	fmt.Printf("%.2f%%\n", float32(count2) / float32(n) * 100)
	fmt.Printf("%.2f%%\n", float32(count3) / float32(n) * 100)
	fmt.Printf("%.2f%%\n", float32(count4) / float32(n) * 100)
	fmt.Printf("%.2f%%\n", float32(count5) / float32(n) * 100)

}
